api/client/src/main/java/org/projectnessie/client/NessieConfigConstants.java

```diff
@@ -244,6 +244,12 @@ public final class NessieConfigConstants {
    */
   public static final String CONF_NESSIE_CLIENT_BUILDER_IMPL = "nessie.client-builder-impl";
 
+  /**
+   * Config property name ({@value #CONF_NESSIE_CLIENT_API_VERSION}) for specifying the client API
+   * version.
+   */
+  public static final String CONF_NESSIE_CLIENT_API_VERSION = "nessie.client-api-version";
```

Comment:
Specifically, I suppose with `nessie.client-api-version` set to `2` we can automatically promote the result of `HttpClientBuilder.build(NessieApiV1.class)` to be a `NessieApiV2`.

Reply:
Isn't there a conflict lurking here between setting the api version with a configuration option vs. setting the api version through the parameter passed to `HttpClientBuilder.build(Class<NessieApi>)`? Imho the proper way to move forward would be to deprecate `HttpClientBuilder.build(Class<NessieApi>)` and create a new `HttpClientBuilder.build()` method the honours both the new and old ways.

Reply:
I think having a `build(Class<NessieApi>)` is still useful for proper type checks on the clients side, especially when the client specifically wants `NessieApiV2`.

My idea was that version-agnostic clients (like the `NessieCatalo` in Iceberg) would call `build(NessieApiV1.class)` since they are compiled to that interface, but use the config property for controlling actual wire format.